capa/main.py

```diff
@@ -264,6 +276,22 @@ def install_common_args(parser, wanted=None):
             help=f"select backend, {backend_help}",
         )
 
+    if "functions" in wanted:
+        parser.add_argument(
+            "--functions",
+            type=lambda s: s.replace(" ", "").split(","),
+            default=[],
+            help="provide a list of comma-separated functions to analyze (static analysis).",
+        )
```

Comment:
To discuss: is this needed besides show-features? if not, should we add it here?

Reply:
@yelhamer what do you think here?

Reply:
@mr-tz I think using `--functions` for both main and show-features would be better, but I feel like that might be a breaking change however and I wouldn't like to delay the merge of this feature into the next release just because of that.

Maybe we could get this PR merged and then open a second PR (with a milestone for the next major release) that uses the `--functions` keyword instead of `-F`/`--function` in show-features, and then document the breaking change there?

Reply:
I like that. Let's remove it from this PR.

Reply:
do you mean remove `-F`/`--function` from show-features.py?

Reply:
I initially understood to remove the functions filter from this PR.

Reply:
Oh, the purpose of this PR is to have that process/function filtering in `show-features.py` applied to capability extractions as well.

I've implemented here (to capa's core) in a more general manner so it can be used by `show-features.py` without having to filter there.

Reply:
Where do you see the functions filter being used and helpful besides in show-features (my original comment here)?

Reply:
good points